docs/user-guide/configure.md

```diff
@@ -1,53 +1,32 @@
 # Configuring
 
-Stylelint expects a configuration object, and looks for one in a:
+Stylelint expects a configuration object.
 
-- `stylelint.config.js` or `.stylelintrc.js` file
-  - Which module system to use depends on your [default module system configuration](https://nodejs.org/api/packages.html#determining-module-system) for Node.js (e.g., `"type": "module"` in `package.json`).
-- `stylelint.config.mjs` or `.stylelintrc.mjs` file using `export default` (ES module)
-- `stylelint.config.cjs` or `.stylelintrc.cjs` file using `module.exports` (CommonJS)
-- `.stylelintrc.json`, `.stylelintrc.yml`, or `.stylelintrc.yaml` file
-- `.stylelintrc` file in JSON or YAML format
-  - We recommend adding an extension (e.g., `.json`) to help your editor provide syntax checking and highlighting.
-- `stylelint` property in `package.json`
-
-ES module example:
-
-```js
-/** @type {import('stylelint').Config} */
-export default {
-  rules: {
-    "block-no-empty": true
-  }
-};
-```
+Starting from the current working directory, Stylelint searches upwards until it finds a `stylelint.config.{js,mjs}` file that exports one. Or, you can use the [`--config`](cli.md#--config--c) or [`configFile`](options.md#configfile) option to short-circuit the search.
 
 > [!NOTE]
-> The `@type` JSDoc annotation enables Typescript to autocomplete and type-check.
+> Stylelint currently supports other configuration formats, but we may remove these in the future:
+>
+> - `.stylelintrc` file in JSON or YAML format.
+> - `.stylelintrc.json` file or `stylelint` property in `package.json`
+> - `.stylelintrc.yml` or `.stylelintrc.yaml` file
+> - `.stylelintrc.js` file using `export default` or `module.exports`
+> - `.stylelintrc.mjs` file using `export default`
```

Comment:
Listing `.mjs` and `.js` first might make it more likely that users will pick that. As some users might not have a preference of their own and will just pick the first in the list.